va/va.go

```diff
@@ -251,99 +251,85 @@ type verificationRequestEvent struct {
 	Error             string `json:",omitempty"`
 }
 
+// ipError is a sentinel error type used to pass though the IP address of the
+// remote host when an error occurs during HTTP-01 and TLS-ALPN domain
+// validation.
+type ipError struct {
+	ip  net.IP
+	err error
+}
+
+// Unwrap returns the underlying error.
+func (i ipError) Unwrap() error {
+	return i.err
+}
+
+// Error returns a string representation of the error.
+func (i ipError) Error() string {
+	return i.err.Error()
+}
+
 // detailedError returns a ProblemDetails corresponding to an error
 // that occurred during HTTP-01 or TLS-ALPN domain validation. Specifically it
 // tries to unwrap known Go error types and present something a little more
 // meaningful. It additionally handles `berrors.ConnectionFailure` errors by
 // passing through the detailed message.
-func detailedError(err error, targetIP net.IP) *probs.ProblemDetails {
+func detailedError(err error) *probs.ProblemDetails {
+	var ipErr ipError
+	if errors.As(err, &ipErr) {
+		detailedErr := detailedError(ipErr.err)
+		// Some detailed errors already contain the IP address. When this is the case,
+		// we don't want to add it again.
```

Comment:
I think the only errors that match this description are `net.Errors`, right? How about we match on the type rather than doing a string match?

Reply:
Ah, good idea, I'll check on that.

Reply:
Updating from conversation: while `net.OpError` does include IP addresses, we actually mask those here in `detailedError`, providing a simplified `berrors.ConnectionError` with no IP address.

The errors that match this description are the already-existing ones like `Fetching http://example.com [1.2.3.4]: yadda yadda`. We're going to reformat those so they are consistent with the new format that always includes the IP address at the beginning.